geniza/entities/views.py

```diff
@@ -176,27 +179,105 @@ class PersonListView(ListView):
     # Translators: description of people list/browse page
     page_description = _("Browse people present in Geniza documents.")
     paginate_by = 50
+    form_class = PersonListForm
+    applied_filter_count = 0
+
+    # fields to facet
+    facet_fields = ["gender", "role__name", "persondocumentrelation__type__name"]
 
     def get_queryset(self, *args, **kwargs):
         """modify queryset to sort and filter on people in the list"""
-        # order people by primary name unaccented
-        return (
-            Person.objects.filter(names__primary=True)
-            .annotate(
+        people = (
+            Person.objects.filter(names__primary=True).annotate(
                 name_unaccented=ArrayAgg("names__name__unaccent", distinct=True),
             )
+            # order people by primary name unaccented
             .order_by("name_unaccented")
         )
 
+        form = self.get_form()
+        # bail out if form is invalid
+        if not form.is_valid():
+            return people.none()
+
+        # filter by each supported field
+        search_opts = form.cleaned_data
+        self.applied_filter_count = 0
+        if "gender" in search_opts and search_opts["gender"]:
```

Comment:
a bit more readable and efficient to do this way:
```suggestion
        if search_opts.get("gender"):
```